### PR TITLE
Date Control: Extract stats date control into a reusable component to keep consistency.

### PR DESCRIPTION
**Pull request by vykes-mac**
Related to https://github.com/Automattic/dotcom-forge/issues/9770

## Proposed Changes
As part of the goal to make design across calypso more consistent we are extracting the [global date picker ](puPv3-gtX-p2) used in Jetpack stats into a reusable `DateControl` component that will be able to used in the sites log page while maintaining the styling and behavior.

* Extract the styling and reusable parts of `StatsDateControl` into `DateControl`
* Modify `StatsDateControl` to use the new `DateControl` component while keeping the parts unique to the `StatsDateControl`

image

## Why are these changes being made?

* As part of keeping consistency across calypso 

## Testing Instructions

* Go to `/stats/day/siteSlug` and ensure the date control works as expected and nothing breaks

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/components/stats-date-control/types.d.ts [hunk 1]
```diff
@@ -12,51 +12,4 @@ interface StatsDateControlProps {
...
-}
-
-export {
-	StatsDateControlProps,
-	DateControlPickerProps,
-	DateControlPickerShortcut,
-	DateControlPickerShortcutsProps,
-	DateControlPickerDateProps,
-};
+export { StatsDateControlProps };
```
Comment (zaguiini) [inline client/components/stats-date-control/types.d.ts]: I think we can get rid of this file entirely if we move the props interface to the component's file.

**Review by zaguiini — COMMENTED** [review]
These fonts look a bit bigger compared to the other texts in the page. Can we make them smaller?

image

**Comment by vykes-mac** [comment]
@annacmc @kangzj I see that you've worked on this component before, It seems a lot the classes in the stylesheet file doesn't apply to the component any at all. Do you have any background on this?

**Comment by vykes-mac** [comment]
> These fonts look a bit bigger compared to the other texts in the page. Can we make them smaller?

I wouldn't change it without design input as it would need to be changed in the `data-range` component. @lucasmendes-design do you have any input here? maybe it was a design decision for the fonts to be that size in the component.

**Comment by kangzj** [comment]
> @annacmc @kangzj I see that you've worked on this component before, It seems a lot the classes in the stylesheet file doesn't apply to the component any at all. Do you have any background on this?

These are all safe to remove - we intended to but didn't get the chance yet.

**Comment by annacmc** [comment]
> I see that you've worked on this component before, It seems a lot the classes in the stylesheet file doesn't apply to the component any at all. Do you have any background on this?

Exactly what Jasper said. We were at one point running a previous datepicker in some situations and hadn't yet cleaned out all the code from the old old one. So it's fine to discard anything that doesn't apply to the component at this time.

**Review by kangzj — APPROVED** [review]
Works well and looks good! Thanks for taking on this 👍

**Pull request merged**